src/Microsoft.VisualStudio.SDK.Analyzers/VSSDK001DeriveFromAsyncPackageCodeFix.cs

```diff
@@ -0,0 +1,141 @@
+                initializeMethodSyntax = updatedRoot.GetCurrentNode(initializeMethodSyntax);
+                var newBody = initializeMethodSyntax.Body;
+                if (baseInitializeInvocationSyntax != null)
+                {
+                    var baseInitializeAsyncInvocationBookmark = new SyntaxAnnotation();
+                    var baseInitializeAsyncInvocationSyntax = SyntaxFactory.AwaitExpression(
+                        baseInitializeInvocationSyntax
+                            .WithLeadingTrivia()
+                            .WithExpression(
+                                SyntaxFactory.MemberAccessExpression(
+                                    SyntaxKind.SimpleMemberAccessExpression,
+                                    SyntaxFactory.BaseExpression(),
+                                    SyntaxFactory.IdentifierName(Types.AsyncPackage.InitializeAsync)))
+                            .AddArgumentListArguments(
+                                SyntaxFactory.Argument(cancellationTokenLocalVarName),
+                                SyntaxFactory.Argument(progressLocalVarName)))
+                        .WithLeadingTrivia(baseInitializeInvocationSyntax.GetLeadingTrivia())
+                        .WithAdditionalAnnotations(baseInitializeAsyncInvocationBookmark);
+                    newBody = newBody.ReplaceNode(initializeMethodSyntax.GetCurrentNode(baseInitializeInvocationSyntax), baseInitializeAsyncInvocationSyntax);
+                    var baseInvocationStatement = newBody.GetAnnotatedNodes(baseInitializeAsyncInvocationBookmark).First().FirstAncestorOrSelf<StatementSyntax>();
+
+                    var leadingTrivia = SyntaxFactory.TriviaList(
+                        SyntaxFactory.LineFeed,
+                        SyntaxFactory.Comment(@"// When initialized asynchronously, we *may* be on a background thread at this point."),
+                        SyntaxFactory.LineFeed,
+                        SyntaxFactory.Comment(@"// Do any initialization that requires the UI thread after switching to the UI thread."),
+                        SyntaxFactory.LineFeed,
+                        SyntaxFactory.Comment(@"// Otherwise, remove the switch to the UI thread if you don't need it."),
+                        SyntaxFactory.LineFeed);
+
+                    var switchToMainThreadStatement = SyntaxFactory.ExpressionStatement(
+                        SyntaxFactory.AwaitExpression(
+                            SyntaxFactory.InvocationExpression(
+                                SyntaxFactory.MemberAccessExpression(
+                                    SyntaxKind.SimpleMemberAccessExpression,
+                                    SyntaxFactory.MemberAccessExpression(
+                                        SyntaxKind.SimpleMemberAccessExpression,
+                                        Types.ThreadHelper.TypeSyntax,
+                                        SyntaxFactory.IdentifierName(Types.ThreadHelper.JoinableTaskFactory)),
```

Comment:
Why not use `this.JoinableTaskFactory` here instead? That's what we're planning on doing in the AsyncPackage template when we get to it.

Reply:
oh ya. whoops. Thanks.

Reply:
Given Mads already completed the PR, I have a commit that addresses this that I will include in a future PR.